tests/unittests/ontology_validation_tests.py

```
@@ -116,6 +119,41 @@ def test_rejects_component_invalid_for_type(self):
+        ao = AnnotationOntology(
+            name="test",
+            attributes=[
+                AttributeSpec(
+                    name="vehicle_type",
+                    type="str",
+                    component="radio",
+                    taxonomy="vehicle_classes",
+                ),
+            ],
+        )
+        with self.assertRaises(ValueError):
```

Ruff (0.15.18)

[warning] Use `pytest.raises` instead of unittest-style `assertRaises`

Replace `assertRaises` with `pytest.raises`

(PT027)